src/main/java/uk/gov/pay/ledger/transaction/model/CsvTransactionFactory.java

```diff
@@ -127,7 +127,11 @@ public Map<String, Object> toMap(TransactionEntity transactionEntity) {
                     result.putAll(getFeeBreakdown(transactionEntity, feeBreakdownNode));
                 }
 
-                result.put(FIELD_3D_SECURE_REQUIRED, safeGetAsBoolean(transactionDetails, "requires_3ds", null));
+                Boolean requires3ds = safeGetAsBoolean(transactionDetails, "requires_3ds", null);
+                if (Boolean.FALSE.equals(requires3ds)) {
+                    requires3ds = null;
+                }
+                result.put(FIELD_3D_SECURE_REQUIRED, requires3ds);
```

Comment:
Might this be more concise?

```java
Boolean required3ds = safeGetAsBoolean(transactionDetails, "requires_3ds", null);
result.put(FIELD_3D_SECURE_REQUIRED, required3ds ? true : null);
```

Reply:
Actually, don’t do that! It will blow up with a `NullPointerException` when `required3ds` is null!

I think the best we can do is probably something like this, which is similar to what you originally had:

```java
if (Boolean.TRUE.equals(safeGetAsBoolean(transactionDetails, "requires_3ds", null))) {
    result.put(FIELD_3D_SECURE_REQUIRED, true);
} else {
    result.put(FIELD_3D_SECURE_REQUIRED, null);
}
```

It’s a shame we don’t live in a world where [JEP 455](https://openjdk.org/jeps/455) has shipped because then we could do this:

```java
result.put(FIELD_3D_SECURE_REQUIRED, switch(safeGetAsBoolean(transactionDetails, "requires_3ds", null)) {
    case true -> true;
    case null, default -> null;
};

Reply:
I didn't do that. I did this:
Boolean requires3ds = safeGetAsBoolean(transactionDetails, "requires_3ds", null);
result.put(FIELD_3D_SECURE_REQUIRED, Boolean.TRUE.equals(requires3ds) ? true : null);